pkg/client/chartmuseum/chartmuseumtester.go

```diff
@@ -0,0 +1,229 @@
+package chartmuseum
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/bitnami-labs/charts-syncer/api"
+	"github.com/juju/errors"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	cmRegex         = regexp.MustCompile(`(?m)\/charts\/(.*.tgz)`)
+	username string = "user"
+	password string = "password"
+)
+
+// Metadata in Chart.yaml files
+type Metadata struct {
+	AppVersion string `json:"appVersion"`
+	Name       string `json:"name"`
+	Version    string `json:"version"`
+}
+
+// ChartVersion type
+type ChartVersion struct {
+	Name    string   `json:"name"`
+	Version string   `json:"version"`
+	URLs    []string `json:"urls"`
+}
+
+type httpError struct {
+	status int
+	body   string
+}
+
+// RepoTester allows to unit test each repo implementation
+type RepoTester struct {
+	url      *url.URL
+	username string
+	password string
+	t        *testing.T
+	// Map of chart name to indexed versions, as returned by the charts API.
+	index map[string][]*ChartVersion
+
+	// Whether the repo should load an empty index or not
+	emptyIndex bool
+
+	// index.yaml to be loaded for testing purposes
+	indexFile string
+	// Set to simulate HTTP error responses for specific API calls.
+	ChartsPostError *httpError
+}
+
+//NewTester start a fake HTTP server to handle requests and return a RepoTester object with useful info for testing
+func NewTester(t *testing.T, repo *api.Repo, emptyIndex bool, indexFile string) (*RepoTester, func(), error) {
+	tester := &RepoTester{
+		t:          t,
+		username:   username,
+		password:   password,
+		emptyIndex: emptyIndex,
+		indexFile:  indexFile,
+		index:      make(map[string][]*ChartVersion),
+	}
+	s := httptest.NewServer(tester)
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		return nil, s.Close, errors.Trace(err)
+	}
+	tester.url = u
+	return tester, s.Close, nil
+}
+
+// ServeHTTP implements the the http Handler type
+func (rt *RepoTester) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Check basic auth credentals.
+	username, password, ok := r.BasicAuth()
+	if got, want := ok, true; got != want {
+		rt.t.Errorf("got: %t, want: %t", got, want)
+	}
+	if got, want := username, rt.username; got != want {
+		rt.t.Errorf("got: %q, want: %q", got, want)
+	}
+	if got, want := password, rt.password; got != want {
+		rt.t.Errorf("got: %q, want: %q", got, want)
+	}
+
+	// Handle recognized requests.
+	if base, chart := path.Split(r.URL.Path); base == "/api/charts/" && r.Method == "GET" {
+		rt.ChartGet(w, r, chart)
```

Comment:
please name it the other way around: `GetChart`

Reply:
modifies this and the others to start with a verb